### ISSUE TITLE
(cli): cdkv2 init introduces deprecated cdkv1 flag @aws-cdk/core:enableStackNameDuplicates

### ISSUE THREAD

**Issue by brainstorm** [aws/aws-cdk#14814]
```
jsii.errors.JSIIError: Unsupported feature flag '@aws-cdk/core:enableStackNameDuplicates'. This flag existed on CDKv1 but has been removed in CDKv2. CDK will now behave as the same as when the flag is enabled.
```

Running CDKv2 as of today:

```
% cdk --version
2.0.0-rc.1 (build 983379b)
```

### Reproduction Steps

1. cdk init app --language=python
2. cdk synth
3. Kaboom

```
% cdk synth
jsii.errors.JavaScriptError:
  Error: Unsupported feature flag '@aws-cdk/core:enableStackNameDuplicates'. This flag existed on CDKv1 but has been removed in CDKv2. CDK will now behave as the same as when the flag is enabled.
      at FeatureFlags.isEnabled (/private/var/folders/s8/0tz3dy0d1sbg1_cv0ylkx_m80000gn/T/jsii-kernel-GNs5qD/node_modules/aws-cdk-lib/lib/core/lib/feature-flags.js:42:23)
      at new Stack (/private/var/folders/s8/0tz3dy0d1sbg1_cv0ylkx_m80000gn/T/jsii-kernel-GNs5qD/node_modules/aws-cdk-lib/lib/core/lib/stack.js:86:51)
      at /private/var/folders/s8/0tz3dy0d1sbg1_cv0ylkx_m80000gn/T/tmpzq2skimo/lib/program.js:8154:58
      at Kernel._wrapSandboxCode (/private/var/folders/s8/0tz3dy0d1sbg1_cv0ylkx_m80000gn/T/tmpzq2skimo/lib/program.js:8582:24)
      at Kernel._create (/private/var/folders/s8/0tz3dy0d1sbg1_cv0ylkx_m80000gn/T/tmpzq2skimo/lib/program.js:8154:34)
      at Kernel.create (/private/var/folders/s8/0tz3dy0d1sbg1_cv0ylkx_m80000gn/T/tmpzq2skimo/lib/program.js:7895:29)
      at KernelHost.processRequest (/private/var/folders/s8/0tz3dy0d1sbg1_cv0ylkx_m80000gn/T/tmpzq2skimo/lib/program.js:9479:36)
      at KernelHost.run (/private/var/folders/s8/0tz3dy0d1sbg1_cv0ylkx_m80000gn/T/tmpzq2skimo/lib/program.js:9442:22)
      at Immediate._onImmediate (/private/var/folders/s8/0tz3dy0d1sbg1_cv0ylkx_m80000gn/T/tmpzq2skimo/lib/program.js:9443:46)
      at processImmediate (node:internal/timers:464:21)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/rvalls/dev/umccr/infrastructure/cdk/apps/template_cdkv2/app.py", line 10, in <module>
    TemplateCdkv2Stack(app, "TemplateCdkv2Stack",
  File "/Users/rvalls/.mambaforge/envs/umccr/lib/python3.9/site-packages/jsii/_runtime.py", line 83, in __call__
    inst = super().__call__(*args, **kwargs)
  File "/Users/rvalls/dev/umccr/infrastructure/cdk/apps/template_cdkv2/template_cdkv2/template_cdkv2_stack.py", line 7, in __init__
    super().__init__(scope, construct_id, **kwargs)
  File "/Users/rvalls/.mambaforge/envs/umccr/lib/python3.9/site-packages/aws_cdk/__init__.py", line 13726, in __init__
    jsii.create(Stack, self, [scope, id, props])
  File "/Users/rvalls/.mambaforge/envs/umccr/lib/python3.9/site-packages/jsii/_kernel/__init__.py", line 275, in create
    response = self.provider.create(
  File "/Users/rvalls/.mambaforge/envs/umccr/lib/python3.9/site-packages/jsii/_kernel/providers/process.py", line 344, in create
    return self._process.send(request, CreateResponse)
  File "/Users/rvalls/.mambaforge/envs/umccr/lib/python3.9/site-packages/jsii/_kernel/providers/process.py", line 326, in send
    raise JSIIError(resp.error) from JavaScriptError(resp.stack)
jsii.errors.JSIIError: Unsupported feature flag '@aws-cdk/core:enableStackNameDuplicates'. This flag existed on CDKv1 but has been removed in CDKv2. CDK will now behave as the same as when the flag is enabled.
Subprocess exited with error 1
```

### What did you expect to happen?

CDKv2 init && synth should not introduce incompatible & deprecated CDKv1 flags. Removing that property fixes the issue but it shouldn't be there at all in the first place after `cdk init`.

### What actually happened?

👆🏻 See above in problem repro

### Environment

  - **CDK CLI Version  : 2.0.0-rc.1 (build 983379b)**
  - **Framework Version: CDKv2**
  - **Node.js Version: v16.0.0** 
  - **OS               : Darwin m1.local 20.4.0 Darwin Kernel Version 20.4.0: Thu Apr 22 21:46:41 PDT 2021; root:xnu-7195.101.2~1/RELEASE_ARM64_T8101 arm64 arm64 MacBookPro17,1 Darwin**
  - **Language (Version): Python 3.9.0** 
 
--- 

This is :bug: Bug Report

/cc @reisingerf @alexiswl @victorskl

**Comment by rix0rrr** [comment]
You left out commands in your repro, so this is not what you can exactly.

Does not repro for me when I run this in a clean Docker:

```
$ npm install -g aws-cdk@2.0.0-rc.1
$ mkdir banaan
$ cd banaan
$ cdk init app --language=python
$ pip install -r requirements.txt 
$ cdk synth
(synths fine)
```

Are you sure you didn't accidentally run `npx cdk init ...` ?

**Comment by brainstorm** [comment]
Sure, check the full thing:

```
(umccr) rvalls@m1 tmp % mkdir banaan
(umccr) rvalls@m1 tmp % cd banaan
(umccr) rvalls@m1 banaan % cdk init app --language=python
Applying project template app for python

# Welcome to your CDK Python project!

This is a blank project for Python development with CDK.

The `cdk.json` file tells the CDK Toolkit how to execute your app.

This project is set up like a standard Python project.  The initialization
process also creates a virtualenv within this project, stored under the `.venv`
directory.  To create the virtualenv it assumes that there is a `python3`
(or `python` for Windows) executable in your path with access to the `venv`
package. If for any reason the automatic creation of the virtualenv fails,
you can create the virtualenv manually.

To manually create a virtualenv on MacOS and Linux:

```
$ python3 -m venv .venv
```

After the init process completes and the virtualenv is created, you can use the following
step to activate your virtualenv.

```
$ source .venv/bin/activate
```

If you are a Windows platform, you would activate the virtualenv like this:

```
% .venv\Scripts\activate.bat
```

Once the virtualenv is activated, you can install the required dependencies.

```
$ pip install -r requirements.txt
```

At this point you can now synthesize the CloudFormation template for this code.

```
$ cdk synth
```

To add additional dependencies, for example other CDK libraries, just add
them to your `setup.py` file and rerun the `pip install -r requirements.txt`
command.

## Useful commands

 * `cdk ls`          list all stacks in the app
 * `cdk synth`       emits the synthesized CloudFormation template
 * `cdk deploy`      deploy this stack to your default AWS account/region
 * `cdk diff`        compare deployed stack with current state
 * `cdk docs`        open CDK documentation

Enjoy!

Initializing a new git repository...
hint: Using 'master' as the name for the initial branch. This default branch name
hint: is subject to change. To configure the initial branch name to use in all
hint: of your new repositories, which will suppress this warning, call:
hint:
hint: 	git config --global init.defaultBranch <name>
hint:
hint: Names commonly chosen instead of 'master' are 'main', 'trunk' and
hint: 'development'. The just-created branch can be renamed via this command:
hint:
hint: 	git branch -m <name>
Please run 'python3 -m venv .venv'!
Executing Creating virtualenv...
✅ All done!
(umccr) rvalls@m1 banaan % source .venv/bin/activate
(.venv) (umccr) rvalls@m1 banaan % which python
/Users/rvalls/tmp/banaan/.venv/bin/python
(.venv) (umccr) rvalls@m1 banaan % ls
README.md  app.py  banaan  cdk.json  requirements.txt  setup.py  source.bat
(.venv) (umccr) rvalls@m1 banaan % cdk synth
Traceback (most recent call last):
  File "app.py", line 4, in <module>
    import aws_cdk as cdk
ModuleNotFoundError: No module named 'aws_cdk'
Subprocess exited with error 1
(.venv) (umccr) rvalls@m1 banaan % pip install -r requirements.txt
Obtaining file:///Users/rvalls/tmp/banaan (from -r requirements.txt (line 1))
Collecting aws-cdk-lib==2.0.0-rc.4
  Using cached aws_cdk_lib-2.0.0rc4-py3-none-any.whl (51.8 MB)
Collecting constructs<11.0.0,>=10.0.0
  Using cached constructs-10.0.5-py3-none-any.whl (61 kB)
Collecting publication>=0.0.3
  Using cached publication-0.0.3-py2.py3-none-any.whl (7.7 kB)
Collecting jsii<2.0.0,>=1.29.0
  Downloading jsii-1.30.0-py3-none-any.whl (286 kB)
     |████████████████████████████████| 286 kB 1.8 MB/s
Collecting python-dateutil
  Using cached python_dateutil-2.8.1-py2.py3-none-any.whl (227 kB)
Collecting cattrs~=1.5.0
  Using cached cattrs-1.5.0-py3-none-any.whl (19 kB)
Collecting attrs~=20.1
  Using cached attrs-20.3.0-py2.py3-none-any.whl (49 kB)
Collecting typing-extensions~=3.7
  Using cached typing_extensions-3.10.0.0-py3-none-any.whl (26 kB)
Collecting six>=1.5
  Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
Installing collected packages: six, attrs, typing-extensions, python-dateutil, cattrs, publication, jsii, constructs, aws-cdk-lib, banaan
  Running setup.py develop for banaan
Successfully installed attrs-20.3.0 aws-cdk-lib-2.0.0rc4 banaan-0.0.1 cattrs-1.5.0 constructs-10.0.5 jsii-1.30.0 publication-0.0.3 python-dateutil-2.8.1 six-1.16.0 typing-extensions-3.10.0.0
WARNING: You are using pip version 21.1.1; however, version 21.1.2 is available.
You should consider upgrading via the '/Users/rvalls/tmp/banaan/.venv/bin/python3 -m pip install --upgrade pip' command.
(.venv) (umccr) rvalls@m1 banaan % cdk synth
jsii.errors.JavaScriptError:
  Error: Unsupported feature flag '@aws-cdk/core:enableStackNameDuplicates'. This flag existed on CDKv1 but has been removed in CDKv2. CDK will now behave as the same as when the flag is enabled.
      at FeatureFlags.isEnabled (/private/var/folders/s8/0tz3dy0d1sbg1_cv0ylkx_m80000gn/T/jsii-kernel-lYO3Y2/node_modules/aws-cdk-lib/lib/core/lib/feature-flags.js:42:23)
      at new Stack (/private/var/folders/s8/0tz3dy0d1sbg1_cv0ylkx_m80000gn/T/jsii-kernel-lYO3Y2/node_modules/aws-cdk-lib/lib/core/lib/stack.js:86:51)
      at /private/var/folders/s8/0tz3dy0d1sbg1_cv0ylkx_m80000gn/T/tmpmw7g0xhu/lib/program.js:8154:58
      at Kernel._wrapSandboxCode (/private/var/folders/s8/0tz3dy0d1sbg1_cv0ylkx_m80000gn/T/tmpmw7g0xhu/lib/program.js:8582:24)
      at Kernel._create (/private/var/folders/s8/0tz3dy0d1sbg1_cv0ylkx_m80000gn/T/tmpmw7g0xhu/lib/program.js:8154:34)
      at Kernel.create (/private/var/folders/s8/0tz3dy0d1sbg1_cv0ylkx_m80000gn/T/tmpmw7g0xhu/lib/program.js:7895:29)
      at KernelHost.processRequest (/private/var/folders/s8/0tz3dy0d1sbg1_cv0ylkx_m80000gn/T/tmpmw7g0xhu/lib/program.js:9479:36)
      at KernelHost.run (/private/var/folders/s8/0tz3dy0d1sbg1_cv0ylkx_m80000gn/T/tmpmw7g0xhu/lib/program.js:9442:22)
      at Immediate._onImmediate (/private/var/folders/s8/0tz3dy0d1sbg1_cv0ylkx_m80000gn/T/tmpmw7g0xhu/lib/program.js:9443:46)
      at processImmediate (node:internal/timers:464:21)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "app.py", line 10, in <module>
    BanaanStack(app, "BanaanStack",
  File "/Users/rvalls/tmp/banaan/.venv/lib/python3.8/site-packages/jsii/_runtime.py", line 83, in __call__
    inst = super().__call__(*args, **kwargs)
  File "/Users/rvalls/tmp/banaan/banaan/banaan_stack.py", line 7, in __init__
    super().__init__(scope, construct_id, **kwargs)
  File "/Users/rvalls/tmp/banaan/.venv/lib/python3.8/site-packages/aws_cdk/__init__.py", line 13726, in __init__
    jsii.create(Stack, self, [scope, id, props])
  File "/Users/rvalls/tmp/banaan/.venv/lib/python3.8/site-packages/jsii/_kernel/__init__.py", line 275, in create
    response = self.provider.create(
  File "/Users/rvalls/tmp/banaan/.venv/lib/python3.8/site-packages/jsii/_kernel/providers/process.py", line 344, in create
    return self._process.send(request, CreateResponse)
  File "/Users/rvalls/tmp/banaan/.venv/lib/python3.8/site-packages/jsii/_kernel/providers/process.py", line 326, in send
    raise JSIIError(resp.error) from JavaScriptError(resp.stack)
jsii.errors.JSIIError: Unsupported feature flag '@aws-cdk/core:enableStackNameDuplicates'. This flag existed on CDKv1 but has been removed in CDKv2. CDK will now behave as the same as when the flag is enabled.
Subprocess exited with error 1
(.venv) (umccr) rvalls@m1 banaan % cdk --version
2.0.0-rc.4 (build ded111f)
```

**Comment by brainstorm** [comment]
Could you try installing mamba first?: https://github.com/mamba-org/mamba

Perhaps that's the culprit and the reason why you cannot repro this?

**Comment by rix0rrr** [comment]
I wonder if you've also fallen victim to: https://github.com/npm/cli/issues/3196

Which I *think* should have been fixed by this: https://github.com/npm/arborist/commit/2ead1e8588cc0cb424e577de9fc561f47d3eb8ab

Released in `@npm/arborist 2.6.1`, which should be in NPM 7.15.1.

Can you give upgrading to 7.15.1 a shot and then retrying? (Might be worth a clean install as well, just to be sure...)

**Comment by rix0rrr** [comment]
Going to close this as nonreproducible.
